sdk/nodejs/resource.ts

```diff
@@ -149,6 +176,26 @@ export abstract class Resource {
                 opts.protect = opts.parent.__protect;
             }
 
+            for (const parentAlias of opts.parent.__aliases) {
+                if (!opts.aliases) {
+                    opts.aliases = [];
+                }
+
+                let aliasName = name;
+                // If the child name has the parent name as a prefix, then we make the assumption that it was
```

Comment:
Do you think we need to provide a way to "opt out" of this behavior in some way?  I am guessing not because there's no danger in addition some (possibly bogus) aliases?

Reply:
I can't think of any scenario where you would want/need to opt-out of this. You can always add your own aliases and they'll be used first.